### PR TITLE
949 separate logging for callbacks

### LINKED ISSUE #949: Create separate logging for callbacks which run in an external process
They should log to the same graylog, but separate files as we can't access the same file from two processes

### PR DESCRIPTION
**Pull request by dperl-dls**
Fixes #949

Link to dodal PR (if required): #[228](https://github.com/DiamondLightSource/dodal/pull/228)

### To test:
1. Confirm tests pass

### CODE REVIEW

**Comment by dperl-dls** [comment]
Good point - do you think we should separate the repositories at some stage? If so I'll move the loggers to be with the modules

File: src/hyperion/log.py [hunk 1]
```diff
@@ -30,36 +38,38 @@ def set_dcgid_tag(dcgid):
...
 
 
-def _get_logging_file_path() -> Path:
+def _get_logging_file_path(filename: str) -> Path:
     """Get the path to write the hyperion log files to.
 
     If the HYPERION_LOG_DIR environment variable exists then logs will be put in here.
-
-    If no envrionment variable is found it will default it to the tmp/dev directory.
+    If no envrionment variable is found it will default it to the ./tmp/dev directory.
```
Comment (DominicOram) [inline src/hyperion/log.py:67]: Nit: Spelling on environment

File: src/hyperion/log.py [hunk 2]
```diff
@@ -30,36 +38,38 @@ def set_dcgid_tag(dcgid):
...
     """
-    logging_path: Path
-
-    hyperion_log_dir = environ.get("HYPERION_LOG_DIR")
-    if hyperion_log_dir:
-        logging_path = Path(hyperion_log_dir)
-    else:
-        logging_path = Path("./tmp/dev/")
-
+    logging_path = Path(environ.get("HYPERION_LOG_DIR") or "./tmp/dev/")
```
Comment (DominicOram) [inline src/hyperion/log.py:73]: This is beautiful, thanks!

File: src/hyperion/unit_tests/test_log/test_log.py [hunk 3]
```diff
@@ -96,3 +101,44 @@ def test_messages_logged_from_dodal_and_hyperion_get_sent_to_graylog_and_file(
...
+
+    assert len(total_filehandler_calls) == len(total_graylog_calls)
+
+    hyperion_filehandler = hyperion_handlers[2]
+    ispyb_filehandler = ispyb_logger.handlers[2]
+    nexus_filehandler = nexus_logger.handlers[2]
+
+    assert nexus_filehandler.baseFilename != hyperion_filehandler.baseFilename  # type: ignore
+    assert ispyb_filehandler.baseFilename != hyperion_filehandler.baseFilename  # type: ignore
+    assert ispyb_filehandler.baseFilename != nexus_filehandler.baseFilename  # type: ignore
```
Comment (DominicOram) [inline src/hyperion/unit_tests/test_log/test_log.py:144]: Nit: I would maybe do an explicit cast to `FileHandler` but not that big a deal

**Review by DominicOram — APPROVED** [review]
Great, thanks for cleaning up the types as you went too. Nits in code, take them or leave them

File: src/hyperion/log.py [hunk 4]
```diff
@@ -30,36 +38,38 @@ def set_dcgid_tag(dcgid):
 
 
 def set_up_logging_handlers(
-    logging_level: Union[str, None] = "INFO", dev_mode: bool = False
+    logger=dodal_logger,
+    logging_level: Union[str, None] = "INFO",
+    dev_mode: bool | None = False,
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/hyperion/log.py]: Now obselete

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**